### PR TITLE
Truncating fields that are sent to slack

### PR DESCRIPTION
**Pull request by prashbnair**
Signed-off-by: Prashant Balachandran <pnair@redhat.com>

Fixed issue https://github.com/prometheus/alertmanager/issues/2769

### CODE REVIEW

**Comment by roidelapluie** [comment]
Thanks for your contribution. I think we should limit the length of the fields we send to slack,  like we do for pageduty, instead of limiting the number of alerts.

**Comment by prashbnair** [comment]
Removed the max alerts config parameter and truncated the field that is the longest as per https://github.com/prometheus/alertmanager/blob/main/template/default.tmpl#L15

File: notify/slack/slack.go [hunk 1]
```diff
@@ -92,13 +93,22 @@ func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
 		tmplText = notify.TmplText(n.tmpl, data, &err)
 	)
 	var markdownIn []string
+	key, err := notify.ExtractGroupKey(ctx)
+
+	if err != nil {
+		return false, err
+	}
```
Comment (simonpasquier) [inline notify/slack/slack.go:100]: i would move this block inside the `if truncated { ... }` block.

Comment (prashbnair) [reply notify/slack/slack.go]: Done

File: notify/slack/slack.go [hunk 2]
```diff
@@ -92,13 +93,22 @@ func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
...
 	} else {
 		markdownIn = n.conf.MrkdwnIn
 	}
+	title, truncated := notify.Truncate(tmplText(n.conf.Title), 1024)
+	if truncated {
+		key, err := notify.ExtractGroupKey(ctx)
+		if err != nil {
+			return false, err
+		}
+		level.Debug(n.logger).Log("msg", "Truncated text", "text", title, "key", key)
```
Comment (simonpasquier) [inline notify/slack/slack.go:108]:
```suggestion
		level.Debug(n.logger).Log("msg", "Truncated title", "text", title, "key", key)
```

Comment (prashbnair) [reply notify/slack/slack.go]: Done

**Pull request merged**